tutorials/flume-capture-events.md

```diff
@@ -0,0 +1,179 @@
+
+```JSP
+<html>
+  <head>
+    <title>Kite Example</title>
+  <head>
+  <body>
+    <h2>Kite Example</h2>
+    <form name="input" action="send" method="get">
+        User ID: <input type="text" name="user_id" value="1">
+        Message: <input type="text" name="message" value="Hello!">
+        <input type="submit" value="Send">
+    </form>
+  </body>
+</html>
+```
+
+## LoggingServlet
+
+When you submit a message from the JSP, the LoggingServlet receives and processes the request. The following is mostly standard servlet code, with some notes about application-specific snippets.
+
+```Java
+package org.kitesdk.examples.demo;
+```
+
+The servlet parses information from the request to create a StandardEvent object. However, you won't find any source code for `org.kitesdk.data.event.StandardEvent`. During the Maven build, the avro-maven-plugin runs before the compile phase. Any `.avsc` file in the `/main/avro` folder is defined as a Java class. The autogenerated classes have  the methods required to build corresponding Avro `SpecificRecord` objects of that type. `SpecificRecord` objects permit efficient access to object fields.
+
+```Java
+
+import org.kitesdk.data.event.StandardEvent;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+```
+
+This example sends Log4j messages directly to the Hive data sink via Flume.
+
+```Java
+import org.apache.log4j.Logger;
+
+public class LoggingServlet extends HttpServlet {
+
+  private final Logger logger = Logger.getLogger(LoggingServlet.class);
+
+  @Override
+  protected void doGet(HttpServletRequest request, HttpServletResponse
+      response) throws ServletException, IOException {
+
+    response.setContentType("text/html");
+```    
+
+Create a PrintWriter instance to write the response page.
+
+```Java
+  PrintWriter pw = response.getWriter();
+
+    pw.println("<html>");
+    pw.println("<head><title>Kite Example</title></head>");
+    pw.println("<body>");
+```
+
+Get the user ID and message values from the servlet request.
+
+```Java
+    String userId = request.getParameter("user_id");
+    String message = request.getParameter("message");
+```
+
+If there's no message, don't create a log entry.
+
+```Java
+    if (message == null) {
+      pw.println("<p>No message specified.</p>");
+
+```
+
+Otherwise, print the message at the top of the page body.
+
+```Java
+    } else {
+      pw.println("<p>Message: " + message + "</p>");
+      
+```
+
+Create a new StandardEvent builder.
+
+```Java
+      StandardEvent event = StandardEvent.newBuilder()
+```
+The event initiator is a user on the server. The event is a web message. These can be set as string literals, because the event initiator and event name are always the same.
```

Comment:
I agree that the event generator is orthogonal to the discussion. It would be helpful if https://github.com/kite-sdk/kite-examples/pull/23 were addressed, to ensure that the code is sound, or at least not embarrassing.